app/controllers/mixins/middleware_common_mixin.rb

```diff
@@ -25,4 +33,89 @@ def clear_topology_breadcrumb
       @breadcrumbs.clear
     end
   end
+
+  def init_show(model_class = self.class.model)
+    clear_topology_breadcrumb
+    @lastaction = 'show'
+    @gtl_url = '/show'
+    @ems = @record = identify_record(params[:id], model_class)
+    return false if record_no_longer_exists?(@record)
```

Comment:
perhaps lines 40-41 should be the first things in this method? so if there's no record, no point in clearing breadcrumbs, etc. just wondering, but maybe this was always the the order of things.